src/NetAnalyzers/Core/Microsoft.NetCore.Analyzers/Runtime/BufferBlockCopyLengthAnalyzer.cs

```diff
@@ -0,0 +1,166 @@
+                   .GetMembers("BlockCopy")
+                   .OfType<IMethodSymbol>()
+                   .FirstOrDefault();
+
+                if (blockCopyMethod is null)
+                {
+                    return;
+                }
+
+                if (!context.Compilation.TryGetOrCreateTypeByMetadataName(WellKnownTypeNames.SystemArray, out INamedTypeSymbol? arrayType))
+                {
+                    return;
+                }
+
+                IPropertySymbol arrayLengthProperty = arrayType
+                    .GetMembers("Length")
+                    .OfType<IPropertySymbol>()
+                    .FirstOrDefault();
+
+                context.RegisterOperationAction(context =>
+                {
+                    var invocationOperation = (IInvocationOperation)context.Operation;
+
+                    if (!invocationOperation.TargetMethod.Equals(blockCopyMethod))
+                    {
+                        return;
+                    }
+
+                    ImmutableArray<IArgumentOperation> arguments = IOperationExtensions.GetArgumentsInParameterOrder(invocationOperation.Arguments);
+
+                    if (arguments.Length != 5)
+                    {
+                        return;
+                    }
+
+                    if (arguments[0].Parameter.Type.SpecialType != SpecialType.System_Array ||
+                        arguments[1].Parameter.Type.SpecialType != SpecialType.System_Int32 ||
+                        arguments[2].Parameter.Type.SpecialType != SpecialType.System_Array ||
+                        arguments[3].Parameter.Type.SpecialType != SpecialType.System_Int32 ||
+                        arguments[4].Parameter.Type.SpecialType != SpecialType.System_Int32)
+                    {
+                        return;
+                    }
+
+                    // Buffer.BlockCopy(Array src, int srcOffset, Array dst, int dstOffset, int count)
+                    IArgumentOperation sourceArgument = arguments[0];
+                    IArgumentOperation destinationArgument = arguments[2];
+                    IArgumentOperation countArgument = arguments[4];
+
+                    bool CheckArrayLengthLocalReference(IArgumentOperation targetArgument, IPropertyReferenceOperation lengthPropertyArgument)
+                    {
+                        if (targetArgument.Value is IConversionOperation targetArgValue)
+                        {
+                            if (lengthPropertyArgument.Instance.GetReferencedMemberOrLocalOrParameter() == targetArgValue.Operand.GetReferencedMemberOrLocalOrParameter())
+                            {
+                                IArrayTypeSymbol countArgumentArrayTypeSymbol = (IArrayTypeSymbol)lengthPropertyArgument.Instance.Type;
+                                if (countArgumentArrayTypeSymbol.ElementType.SpecialType != SpecialType.System_Byte && countArgumentArrayTypeSymbol.ElementType.SpecialType != SpecialType.System_SByte)
```

Comment:
I think bool type should also be excluded

Reply:
You're right that `bool` can be excluded here. https://github.com/dotnet/runtime/issues/33765 mostly talked about byte and `sbyte`, that's why @mahdiva dealt with those first. What do you think about filing an issue to fix the implementation here to exclude bool and getting this merged in the meanwhile?

Reply:
Filed https://github.com/dotnet/roslyn-analyzers/issues/5353 optimistically :) Please feel free to disagree if you think we should fix it in this PR itself!

Reply:
> What do you think about filing an issue to fix the implementation here to exclude bool and getting this merged in the meanwhile?

I support that approach

Reply:
> What do you think about filing an issue to fix the implementation here to exclude bool and getting this merged in the meanwhile?

I am OK with that, but i assumed it needs very little change like just need another condition for bool type, no? 
```if (countArgumentArrayTypeSymbol.ElementType.SpecialType != SpecialType.System_Byte && countArgumentArrayTypeSymbol.ElementType.SpecialType != SpecialType.System_SByte && countArgumentArrayTypeSymbol.ElementType.SpecialType != SpecialType.System_Bool)```

Reply:
> Anyway I am OK with resolving it later too

But this needs to be resolved within 6.o as it would cause false positive

Reply:
Created PR #5373 that fixes this issue